eng/pipelines/templates/stages/archetype-android-release.yml

```diff
@@ -19,36 +19,72 @@ stages:
   - stage: Signing
     dependsOn: ${{parameters.DependsOn}}
     jobs:
-      - deployment: SignPackage
-        environment: esrp
+      - job: SignPackage
         timeoutInMinutes: 20
         pool:
-          name: azsdk-pool-mms-ubuntu-2004-general
-          image: azsdk-pool-mms-ubuntu-2004-1espt
-          os: linux
-
-        strategy:
-          runOnce:
-            deploy:
-              steps:
-                - checkout: self
-
-                - checkout: azure-sdk-build-tools
-                  path: azure-sdk-build-tools
-
-                - download: current
-                  artifact: ${{parameters.ArtifactName}}
-                  timeoutInMinutes: 5
-
-                - template: /tools/java-esrp-signing/java-esrp-signing.yml@azure-sdk-build-tools
-                  parameters:
-                    Artifacts: ${{parameters.Artifacts}}
-                    ArtifactDirectory: $(Pipeline.Workspace)/${{parameters.ArtifactName}}
-                
-                - template: /eng/common/pipelines/templates/steps/publish-1es-artifact.yml
-                  parameters:
-                    ArtifactName: ${{parameters.ArtifactName}}-signed
-                    ArtifactPath: $(Pipeline.Workspace)/${{parameters.ArtifactName}}
+          name: azsdk-pool-mms-win-2022-general
+          image: azsdk-pool-mms-win-2022-1espt
+          os: windows
+        steps:
+          - checkout: self
+
+          - checkout: azure-sdk-build-tools
+            path: azure-sdk-build-tools
+
+          - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
```

Comment:
Either this or `checkout: self` should be sufficient I think, together they are redundant.

Reply:
No, unfortunately, they're both necessary. These were lifted directly from the [original publish step](https://github.com/Azure/azure-sdk-for-android/pull/1832/files/22f06481580ab6edc5c9536a0d5840690eefd3f6#diff-2876c69c83479f9fb61da557dcf20fe4049d06f98023a3743b63b2a417fdb626L109).